src/java.base/share/native/libjava/ClassLoader.c

```diff
@@ -287,9 +287,12 @@ Java_java_lang_ClassLoader_findBootstrapClass(JNIEnv *env, jobject loader,
         JNU_ThrowOutOfMemoryError(env, NULL);
         return NULL;
     }
-    fixClassname(clname);
-
-    if (!verifyClassname(clname, JNI_TRUE)) {  /* expects slashed name */
+    // disallow slashes in input, change '.' to '/'
+    if (verifyFixClassname(clname)) {
```

Comment:
perhaps we should replace all use of `fixClassname` with `verifyFixClassname` and remove `fixClassname`.

Reply:
This suggestion makes sense to me.  verifyClassName is only used once in Class.c passing false so you could remove that argument.
It's hard to see how fixClassName then verifyClassname is equivalent to verifyFixClassname but verifyFixClassname makes more sense than verifyClassname.
I think this return:
    return (p != 0 && p - name == (ptrdiff_t)length);
implies a non-utf8 character was found?

Reply:
Actually I think replacing fixClassName with verifyFixClassname will be awkward since the latter returns a value that's not checked in all the callers of fixClassName.  Maybe you could write fixClassName as:
void fixClassName() { verifyFixClassName(); with some assertion it passed? }